### DISCUSSION TITLE
Symbol file structure

### DISCUSSION THREAD

**Discussion by zsolt-rozsnyai** [slajerek/RetroDebugger#9] [Q&A]
Hello @slajerek !

First of all many thanks for this great project, it is very useful and makes my life easier a lot while developing for C64.

I am working on integrating Retro Debugger with my assembler and I want to start RD up with my labels, breakpoints and watches generated in the assembler. Meanwhile I just can't figure out the file format to inject these information. I can upload them manually as I can generate the files for importing but I want to automate this process as it is pretty tedious to load these files after every change. I was looking at the code but TBH I am not so familiar with c++ and could not figure it out by myself. Could you give example files for the `-symbols` and `-breakpoints` command line options?

Thanks in advance!

**Comment by slajerek** [comment]
Hello @zsolt-rozsnyai 

Apologies, I totally overlooked this message. There's some documentation, I guess you would like to look at KickAssembler integration specs we did with Mads, author of the KickAssembler:

https://github.com/slajerek/RetroDebugger/tree/master/docs/specs

File format of code labels, data watches and breakpoints is straightforward. It is stored as Hjson (https://hjson.github.io), but not documented yet. If you need more details, please check `Serialize` and `Deserialize` methods in these classes:

https://github.com/slajerek/RetroDebugger/blob/master/src/DebugInterface/Symbols/CDebugSymbolsCodeLabel.cpp
https://github.com/slajerek/RetroDebugger/blob/master/src/DebugInterface/Symbols/CDebugSymbolsDataWatch.cpp

and also `SerializeBreakpointsToHjson`, that will serialize all segments in
https://github.com/slajerek/RetroDebugger/blob/master/src/DebugInterface/Symbols/CDebugSymbols.cpp

and then `SerializeBreakpoints`, that will serialize one segment:
https://github.com/slajerek/RetroDebugger/blob/master/src/DebugInterface/Symbols/CDebugSymbolsSegment.cpp

I suggest just store some of watches, labels, breakpoints by selecting Code/C64 Labels/Export labels, etc. and hope you'll get the idea. In case of problems let me know.

**Comment by LowPowerCell** [comment]
I have a question.

I am working with C64 Studio. When starting the program, a TMP file is created under
User/.../AppData/Local/Temp/... .tmp.

This file contains, among other things, the following label definitions:

add_label $376A .standard_vorgaben_variable
add_label $3789 .next_level_counter
add_label $378A .next_level_vorgaben_variable
add_label $37AB .next_level_counter_loop
add_label $0001 .Cursor_farbe

exit


In 64Debugger, these labels are displayed correctly, even though the addresses always end with an A.
However, this does not work in RetroDebugger the labels are not shown there.

Why is that?
Otherwise, I would have to manually convert this file every time to make the labels visible.
In general, I find RetroDebugger quite good, especially because:
- you can adjust the size of the individual memory fields
- the 1541 disk drive memory is displayed correctly

Is there a way to automatically re-import the labels from the original program so that they are visible immediately after loading?
